src/modules/Builder/features/ReportConfigSetting/ReportConfigSetting.hooks.test.ts

```diff
@@ -0,0 +1,227 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import { act, renderHook } from '@testing-library/react';
+import { useParams } from 'react-router-dom';
+
+import { SingleApplet } from 'shared/state';
+import { authStorage } from 'shared/utils/authStorage';
+
+import { verifyReportServer, setPasswordReportServer } from './ReportConfigSetting.utils';
+import { useCheckReportServer, useDefaultValues } from './ReportConfigSetting.hooks';
+import {
+  OK_MESSAGE,
+  SUCCESS_MESSAGE,
+  defaultValues as initialValues,
+} from './ReportConfigSetting.const';
+
+jest.mock('react-router-dom', () => ({
+  useParams: jest.fn(),
+}));
+
+jest.mock('shared/utils/authStorage', () => ({
+  authStorage: {
+    getAccessToken: jest.fn(),
+  },
+}));
+
+jest.mock('./ReportConfigSetting.utils', () => ({
+  verifyReportServer: jest.fn(),
+  setPasswordReportServer: jest.fn(),
+}));
+
+describe('useCheckReportServer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('onVerify returns true when server responds with OK_MESSAGE', async () => {
+    const mockUrl = 'http://example.com';
```

Comment:
To reduce a bit of code, could you please move the same code in top of the file and the same logic in a function, for ex.:
```
const mockUrl = 'http://example.com';
const mockPublicKey = 'mockPublicKey';
const mockToken = 'mockToken';

const renderReportServerHook = (message: string) => {

 (authStorage.getAccessToken as jest.Mock).mockReturnValue(mockToken);
    (verifyReportServer as jest.Mock).mockResolvedValue({
      json: async () => ({ message }),
    });
    (useParams as jest.Mock).mockReturnValue({ appletId: 'testAppletId', ownerId: 'testOwnerId' });

    return renderHook(() =>
      useCheckReportServer({ url: mockUrl, publicKey: mockPublicKey }),
    );
}
```